src/Event/DoctrineDatasetListener.php

```diff
@@ -103,7 +104,8 @@ public function postUpdate(Dataset $dataset, LifecycleEventArgs $args): void
                 ) or
                 ($entity->getDif() instanceof DIF and $entity->getIdentifiedStatus() === DIF::STATUS_APPROVED)
             ) {
-                $this->publisher->publish($dataset->getId(), RabbitPublisher::DOI_PRODUCER, 'doi');
+                $doiMessage = new DoiMessage($entity->getId(), DoiMessage::ISSUE_OR_UPDATE);
```

Comment:
Unsure what $dataset is and $entity is (if the Entity is Dataset).

Reply:
In this scenario, $entity and $dataset should be Dataset